pkg/controller/jwks/jwks.go

```diff
@@ -52,8 +52,25 @@ type Controller struct {
 func (c *Controller) HandleIndex() http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		ctx := r.Context()
+
```

Comment:
@jeremyfaller can you take a look at this? There was a subtle bug that would result in a deadlock if the same cacher was given to the public key cache and the jwk builder. I think I fixed it, but would appreciate a second look.

Reply:
Not sure I completely see the deadlock, would be great to have a test to prove it's fixed. In any case:

AFA the JWK stuff, this separation looks like it makes sense.